tensorizer/tensorizer.py

```diff
@@ -359,22 +365,81 @@ def seek(self, position, whence=SEEK_SET):
             raise (Exception("Seeking is unsupported"))
 
 
+def save_to_state_dict(m, destination, prefix, keep_vars):
+    r"""Saves module state to `destination` dictionary, containing a state
+    of the module, but not its descendants. This is called on every
+    submodule in :meth:`~torch.nn.Module.state_dict`.
+
+    In rare cases, subclasses can achieve class-specific behavior by
+    overriding this method with custom logic.
+
+    Args:
+        destination (dict): a dict where state will be stored
+        prefix (str): the prefix for parameters and buffers used in this
+            module
+    """
+    for name, param in m._parameters.items():
+        destination[prefix + name] = param if keep_vars else param.detach()
+    for name, buf in m._buffers.items():
+        if name not in m._non_persistent_buffers_set:
+            destination[prefix + name] = buf if keep_vars else buf.detach()
+    extra_state_key = prefix + "_extra_state"
+    if (
+        getattr(m.__class__, "get_extra_state", torch.nn.Module.get_extra_state)
+        is not torch.nn.Module.get_extra_state
+    ):
+        destination[extra_state_key] = m.get_extra_state()
+
+
+def state_dict(
+    m: torch.nn.Module, destination=None, prefix="", keep_vars=True
+) -> OrderedDict:
+    if destination is None:
+        destination = OrderedDict()
+        destination._metadata = OrderedDict()
+    destination._metadata[prefix[:-1]] = local_metadata = dict(
+        version=m._version
+    )
+    save_to_state_dict(m, destination, prefix, keep_vars)
+    # m._save_to_state_dict(destination, prefix, keep_vars)
+    for name, module in m._modules.items():
+        state_dict(
+            module, destination, prefix + name + ".", keep_vars=keep_vars
+        )
+    for hook in m._state_dict_hooks.values():
+        hook_result = hook(m, destination, prefix, local_metadata)
+        if hook_result is not None:
+            destination = hook_result
+    return destination
+
+
 class GooseTensorizer:
     """
     Given a file-like object, either for read or write, serialize tensors
     to the file, or deserialize the tensors.
     """
 
-    def __init__(self, file_obj, compress_tensors=False) -> None:
-        self._file = file_obj
+    def __init__(
+            self,
+            file_obj: Union[io.BufferedIOBase, io.RawIOBase, str],
+            compress_tensors=False) -> None:
+        if isinstance(file_obj, str):
+            self._file = open(file_obj, "rb+")
```

Comment:
The IO mode should be replaced with `"ab+"` which would let the Tensorizer create a new file when needed instead of throwing a `FileNotFound` error if the file isn't there.

Reply:
Fixed. :+1: